File: backend/backend/Models/Quote.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using backend.Enum;

namespace backend.Models
{
    public class Quote
    {
        public int Id { get; set; }

        [Required]
        public int FreelancerId { get; set; }

        [ForeignKey("FreelancerId")] 
        public Freelancer Freelancer { get; set; }


        [Required]
        [Range(0, double.MaxValue)]
        public decimal Price { get; set; }

        [MaxLength(1000)]
        public string? Comment { get; set; }

        [Required]
        public int JobId { get; set; }
        
        [ForeignKey("JobId")]
        public Job Job { get; set; }

        public QuoteState QuoteState { get; set; }

        public int? EvaluationScore { get; set; }
        public string? EvaluationDecision { get; set; }

    }
}

File: frontend/src/assets/models/Quote.ts
import { Freelancer } from "./Freelancer";
import { Job } from "./Job";

export interface Quote {
  id: number;
  freelancerId: number;
  freelancer: Freelancer;
  freelancerUsername: string;
  jobTitle: string;
  price: number;
  comment?: string;
  jobId: number;
  job: Job;
  evaluationScore?: number;
  decision?: string;
  quoteState: number;
  username: string;

}


export function getQuoteState(quoteState: number): string {
    switch (quoteState) {
      case 0: return "Rejected";
      case 1: return "Pending";
      case 2: return "Approved";
      case 3: return "Canceled";
      default: return "Unknown";
    }
  }
